src/utils/Cypher.ts

```diff
@@ -145,4 +132,17 @@ export default class Cypher {
       }
     }
   }
+
+  private static async getCryptoSetting(tenantID: string): Promise<CryptoSetting> {
+    const cryptoSettings = (await SettingStorage.getCryptoSettings(tenantID)).crypto;
+    if (!cryptoSettings) {
+      throw new BackendError({
+        source: Constants.CENTRAL_SERVER,
+        module: MODULE_NAME,
+        method: 'getCryptoSetting',
+        message: `Tenant with ID: ${tenantID} does not have crypto settings.`
```

Comment:
You can get the name with `TenantStorage.getTenant(tenantID).name`. The tenantID is not really a human readable output to log.

Reply:
Yep but a call in the DB just to get the name...
Type to pass the tenant object instead the ID.

Reply:
@LucasBrazi06 not sure what you mean by passing the tenant object?

Reply:
To replace this:
`private static async getCryptoSetting(tenantID: string): Promise<CryptoSetting> {`
by this:
`private static async getCryptoSetting(tenant: Tenant): Promise<CryptoSetting> {`

But this method is only containing DB access so it's useless, DB method should be called right away.

Reply:
But there's no need to call tenant storage unless there's an error thrown. If the crypto setting is there and we call tenant storage to get tenant it is an unnecessary call to db.

Reply:
This can be handled outside this method I think.

Reply:
> But there's no need to call tenant storage unless there's an error thrown. If the crypto setting is there and we call tenant storage to get tenant it is an unnecessary call to db.

Just revert the change I've requested (for nothing) to the previous state: tenantID in the error message. That will be enough and do not require extra DB query.

Reply:
Okay, back to just tenant id now